packages/engine/Source/Shaders/Builtin/Functions/clipPolygons.glsl

```diff
@@ -0,0 +1,46 @@
+float getSignedDistance(vec2 uv, highp sampler2D clippingDistance) {
+    float signedDistance = texture(clippingDistance, uv).r;
+    return (signedDistance - 0.5) * 2.0;
+}
+
+void czm_clipPolygons(highp sampler2D clippingDistance, vec2 clippingDistanceTextureDimensions, int polygonsLength, highp vec2 clippingPosition, int polygonIndex) {
+    // Position is completely outside of polygons bounds
+    vec2 rectUv = clippingPosition;
+    if (rectUv.x <= 0.0 || rectUv.y <= 0.0 || rectUv.x >= 1.0 || rectUv.y >= 1.0) {
+        #ifdef CLIPPING_INVERSE 
+            discard;
+        #endif
+        return;
+    }
+
+    vec2 sampleOffset = max(1.0 / clippingDistanceTextureDimensions, vec2(0.005));
+    float dimension = max(ceil(log2(float(polygonsLength))), float(polygonsLength));
```

Comment:
Whoops, it should only be this case when polygonLength <= 2. Otherwise it's always `ceil(log2(float(polygonsLength))`. This is needed to get the x-dimension of an x-by-x grid that fits the number of polygons.

Fixed here and in `PolygonSignedDistanceFS.glsl`.